src/index.js

```diff
@@ -1,36 +1,103 @@
-    completed: true,
-    index: 1,
-  }, {
-    description: 'complete To Do list project',
-    completed: true,
-    index: 2,
-  },
-  {
-    description: 'wash my car',
-    completed: true,
-    index: 3,
-  },
-];
-const showTasks = (task) => {
-  listArray.sort((a, b) => a.index - b.index);
-  todoList.innerHTML += `<div class="todo" id="todo-con">
-                          <div class="left-side">
-                            <input type="checkbox" id="todo" name="todo" value="">
-                            <p>${listArray[task].description}</p>
-                          </div>
-                          <div class="right-side">
-                            <i class="fa-solid fa-ellipsis-vertical"></i>
-                            <i class="fa-regular fa-trash-can trash-can"></i>
-                          </div>
-                        </div>`;
+    // Function to handle edit action
+    const handleEdit = (index) => {
+      // Get the task description element
+      const taskDescEl = document.getElementsByClassName('task-input-field')[index];
+      // Replace the task description element with an input field
+      const newInputEl = document.createElement('input');
+      newInputEl.setAttribute('type', 'text');
+      newInputEl.setAttribute('value', taskDescEl.innerText);
+      newInputEl.classList.add('edit-input');
+      taskDescEl.replaceWith(newInputEl);
+      // Focus on the input field
+      newInputEl.focus();
+      // Add event listener to the input field to handle the edit action
+      newInputEl.addEventListener('keydown', (event) => {
+        if (event.key === 'Enter') {
+          // Save the edited todo
+          todos[index].description = newInputEl.value;
+          toLocalStorage();
+          renderTodos();
+        }
+      });
+    };
+
+    // Add event listener to each edit icon
+    const editIcon = document.getElementsByClassName('edit');
+    for (let i = 0; i < editIcon.length; i += 1) {
+      editIcon[i].addEventListener('click', () => {
+        handleEdit(i);
+      });
+    }
+    // Function to handle delete action
+    const handleDelete = (index) => {
+      todos.splice(index, 1);
+      toLocalStorage();
+      renderTodos();
+    };
+    // Add event listener to each delete icon
+    const deleteIcon = document.getElementsByClassName('delete');
+    for (let i = 0; i < deleteIcon.length; i += 1) {
+      deleteIcon[i].addEventListener('click', () => {
+        handleDelete(i);
+      });
+    }
+  }
 };
-for (let i = 0; i < listArray.length; i += 1) {
-  showTasks(i);
-}
\ No newline at end of file
+
+const saveTodo = () => {
+  const todoValue = todoInput.value;
+  const isEmpty = todoValue === '';
+  if (isEmpty) {
+    todoInput.innerHTML = 'Empty todo';
+  } else {
+    const todo = {
+      description: todoValue,
+      completed: false,
+      index: todos.length,
+    };
+    todos.push(todo);
+    todoInput.value = '';
+    toLocalStorage();
+    renderTodos();
```

Comment:
- [ ] In the requirements of this project, you're required to have the index of the todos represent the length of the array. That means, if the array is having 5 items, you should make all the 5th item in the array has an index of 5 and not 4th(not the normal way array counts). You will achieve this with this syntax `index: todos.length + 1,` 
Currently, the todos index starts counting at 0. Kindly use the format above. 👍🏽

<img width="471" alt="Screenshot 2023-04-08 at 5 17 47 PM" src="https://user-images.githubusercontent.com/50941074/230734701-0e54f4dc-667f-4bb2-8590-b457a28391f9.png">